hilti/toolchain/src/compiler/optimizer.cc

```diff
@@ -81,9 +94,13 @@ class OptimizerVisitor : public visitor::MutatingPreOrder {
     enum class Stage { COLLECT, PRUNE_USES, PRUNE_DECLS };
     Stage _stage = Stage::COLLECT;
     declaration::Module* _current_module = nullptr;
+    const OperatorUses& op_uses;
```

Comment:
I like that this is `const` so users don't accidentally add data with `operator[]`, but let's not make this a const ref since that disables assignment and move operators (we currently don't use them, but still).

Let's instead make this a private pointer member, and provide access to the data. Currently users _should_ check for presence with `count` and can only then get the data with `at`; our accessor could encapsulate that completely, e.g., 

```cpp
    const OperatorUses::mapped_type* uses(const Operator* x) const {
        if ( _op_uses->count(x) == 0 )
            return nullptr;

        return &_op_uses->at(x);
    }

private:
    const OperatorUses* _op_uses;
```